File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l=0
        r=len(numbers)-1
        ans=[]
        while l<r:
            if numbers[l]+numbers[r]<target:
                l+=1
                print("l",l)
            elif numbers[l]+numbers[r]>target:
                r=r-1
                print("r",r)
            else:
                ans.append(l+1)
                ans.append(r+1)
                break
            
        return ans
        

File: Data Structures & Algorithms/two-integer-sum/submission-5.cpp
#include <algorithm>
#include <iostream>
#include <vector>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {   
        vector<int> res;
        vector<int> temp=nums;
        vector<int> ans;
        sort(nums.begin(),nums.end());
        int i=0;
        int j=nums.size()-1;
        while (i<j)
        {
            if (nums[i]+nums[j]<target){
                i++;
            }
            else if (nums[i]+nums[j]>target){
                j--;
            }
            else{
                res.push_back(nums[i]);
                res.push_back(nums[j]);
                break;
            }
        }
        int a=res[0];
        int b=res[1];
        for (int i=0;i<temp.size();i++)
        {
            if (temp[i]==a){
                ans.push_back(i);
            }
            else if (temp[i]==b){
                ans.push_back(i);
            }
        }
        return ans;

    }

};
